benchmarktests/benchmark_targets.go

```diff
@@ -12,13 +12,23 @@ import (
 	"os"
 	"sort"
 	"sync"
+	"time"
 
 	"github.com/hashicorp/go-hclog"
 	"github.com/hashicorp/hcl/v2"
 	"github.com/hashicorp/vault/api"
 	vegeta "github.com/tsenart/vegeta/v12/lib"
 )
 
+// Configuration that applies to all individual tests
+type TopLevelTargetConfig struct {
+	Duration     time.Duration
+	Tests        []*BenchmarkTarget
+	RandomMounts bool
+	Logger       *hclog.Logger
+	Client       *api.Client
+}
+
```

Comment:
I think we want to rethink this struct a bit. While this proposed solution makes a couple of calls inside of `run.go` and `benchmark_targets.go` simpler I think it makes things a bit more cumbersome and unclear when working with individual benchmark tests, as well as we end up passing around more data than needed. We end up passing this config struct down into each individual benchmark test which also includes a slice to references to the other test structs.

I think we would just want this config struct to contain only configuration options such as random_mounts and now the reference to the benchmark duration:
```
// Passthrough parameters from global Benchmark Config
type TopLevelTargetConfig struct {
	Duration     time.Duration
	RandomMounts bool
}
```

That makes the overall changes to each benchmark test simpler but still allows us to pass additional data in the future.

Reply:
yea that makes sense, I was trying to reduce the number of parameters being passed into a function. but youre right, its grouping them and theyre not really related :/